featurebyte/service/base_document.py

```diff
@@ -513,62 +524,65 @@ async def list_documents_as_dict(
                 sort_by=sort_by,
                 page=page,
                 page_size=page_size,
-                user_id=self.user.id,
             )
         except NotImplementedError as exc:
             raise QueryNotSupportedError from exc
         return {"page": page, "page_size": page_size, "total": total, "data": list(docs)}
 
     async def list_documents_as_dict_iterator(
         self,
-        query_filter: QueryFilter,
-        projection: Optional[Dict[str, Any]] = None,
-        page_size: int = DEFAULT_PAGE_SIZE,
+        sort_by: Optional[list[tuple[str, SortDir]]] = None,
         use_raw_query_filter: bool = False,
+        projection: Optional[Dict[str, Any]] = None,
         **kwargs: Any,
     ) -> AsyncIterator[Dict[str, Any]]:
         """
         List documents iterator to retrieve all the results based on given document service & query filter
 
         Parameters
         ----------
-        query_filter: QueryFilter
-            Query filter
-        projection: Optional[Dict[str, Any]]
-            Project fields to return from the query
-        page_size: int
-            Page size
+        sort_by: Optional[list[tuple[str, SortDir]]]
+            Keys and directions used to sort the returning documents
         use_raw_query_filter: bool
             Use only provided query filter
+        projection: Optional[Dict[str, Any]]
+            Project fields to return from the query
         kwargs: Any
-            Additional keyword arguments passed to the list_documents_as_dict
+            Additional keyword arguments
 
         Yields
         ------
         AsyncIterator[Dict[str, Any]]
             List query output
-        """
-        to_iterate, page = True, 1
 
-        while to_iterate:
-            list_results = await self.list_documents_as_dict(
-                page=page,
-                page_size=page_size,
-                query_filter=query_filter,
-                projection=projection,
-                use_raw_query_filter=use_raw_query_filter,
-                **kwargs,
+        Raises
+        ------
+        QueryNotSupportedError
+            If the persistent query is not supported
+        """
+        sort_by = sort_by or [("created_at", "desc")]
+        query_filter = self.construct_list_query_filter(
+            use_raw_query_filter=use_raw_query_filter, **kwargs
+        )
+        try:
+            docs = cast(
+                AsyncIterator[Dict[str, Any]],
+                await self.persistent.get_iterator(
+                    collection_name=self.collection_name,
+                    query_filter=query_filter,
+                    projection=projection,
+                    sort_by=sort_by,
+                ),
             )
-            for doc in list_results["data"]:
-                yield doc
+        except NotImplementedError as exc:
+            raise QueryNotSupportedError from exc
```

Comment:
Curious in what scenarios will NotImplementedError be raised?

Reply:
This is only raised in unit test due to mockmongo not supporting certain operations